resources/generate_book.py

```diff
@@ -659,6 +659,62 @@ def make_book(rm: ResourceManager, i18n: I18n, local_instance: bool = False):
             sealed_barrel_recipe('tfc:barrel/medium_prepared_hide', 'After scraping, $(thing)Scraped Hides$() must be sealed in a $(l:mechanics/barrels)Barrel$() of $(thing)Water$(), for at least eight hours, for one final cleaning before tanning.').anchor('preparing'),
             sealed_barrel_recipe('tfc:barrel/medium_leather', 'Finally, $(thing)Prepared Hides$() must be sealed in a $(l:mechanics/barrels#tannin)Barrel of Tannin$(), which is an acidic chemical compound that helps to convert the hide to $(thing)Leather$(). After another eight hours, you will be able to remove the $(thing)Leather$() from the barrel.').anchor('tanning'),
         )),
+        entry('weaving', 'Weaving', 'tfc:spindle', pages=(
+            text('$(thing)Weaving$() is the process of combining different kinds of string into $(thing)Cloth$(). While the last step of weaving is done in a $(thing)Loom$(), some cloths such as $(thing)Wool$(), obtained from $(l:mechanics/animal_husbandry#wooly_animals)Wooly Animals$(), requires a $(thing)Spindle$() to obtain $(thing)Wool Yarn$() in order to be woven.'),
+            clay_knapping('tfc:clay_knapping/spindle_head', 'The $(thing)Unfired Spindle Head$() is knapped from clay. It can then be $(l:mechanics/heating)fired$() to make a $(thing)Spindle Head$(). To complete the spindle, craft it with a $(thing)Stick$().'),
+            crafting('tfc:crafting/wool_yarn', text_contents='Crafting $(thing)Wool$() with a Spindle yields $(thing)Wool Yarn$().'),
+            crafting('tfc:crafting/wood/acacia_loom', text_contents='The loom is crafted from just $(thing)Lumber$() and a $(thing)Stick$().'),
+            loom_recipe('tfc:loom/wool_cloth', 'The recipe for $(thing)Wool Cloth$() takes 16 $(thing)Wool Yarn$(). Adding to the loom is done with $(item)$(k:key.use)$(). Then, hold down $(item)$(k:key.use)$() to begin working the loom.'),
+            image('tfc:textures/gui/book/tutorial/loom_full.png', text_contents='The loom after being filled with $(thing)Wool Yarn$().'),
+            image('tfc:textures/gui/book/tutorial/loom_working.png', text_contents='The filled loom being worked by holding $(item)$(k:key.use)$(). When complete, $(item)$(k:key.use)$() retrieves the item.'),
```

Comment:
Suggestion here would be to include a proper animated tutorial sequence: meaning, use /tp, take identically positioned screenshots of each step in the sequence:
1) empty
2) after adding the items
3) halfway through weaving
4) finished

The left page can then be a text description of the steps needed.